### PR TITLE
Fixing FOV locks when changed world

### PR DESCRIPTION
**Pull request by patyhank**
Fix https://github.com/GeyserMC/Geyser/issues/3896

### CODE REVIEW

File: core/src/main/java/org/geysermc/geyser/session/GeyserSession.java [hunk 1]
```diff
@@ -1776,6 +1776,11 @@ public void sendAdventureSettings() {
         Collections.addAll(abilityLayer.getAbilitiesSet(), USED_ABILITIES);
 
         updateAbilitiesPacket.getAbilityLayers().add(abilityLayer);
+        // https://github.com/GeyserMC/Geyser/issues/3896 Add Base mode ability layer to fixing FOV locks when changed world
+        if (spectator) {
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/session/GeyserSession.java:1780]: Why not use the spectator check a few lines above, aka https://github.com/GeyserMC/Geyser/blob/master/core/src/main/java/org/geysermc/geyser/session/GeyserSession.java#L1763-L1772?
Additionally, is intended that spectator mode would have both the base and spectator ability layer?
See your own earlier issue/PR: https://github.com/GeyserMC/Geyser/issues/3769

**Review by onebeastchris — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/geysermc/geyser/session/GeyserSession.java [hunk 2]
```diff
@@ -1765,11 +1765,10 @@ public void sendAdventureSettings() {
         }
 
         // https://github.com/GeyserMC/Geyser/issues/3769 Setting Spectator mode ability layer
-        if (spectator) {
-            abilityLayer.setLayerType(AbilityLayer.Type.SPECTATOR);
-        } else {
-            abilityLayer.setLayerType(AbilityLayer.Type.BASE);
-        }
+        abilityLayer.setLayerType(AbilityLayer.Type.SPECTATOR);
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/session/GeyserSession.java:1768]: Now we would be setting the spectator layer for all gamemodes..
The check is still needed, but there was no need for two of them

**Review by onebeastchris — CHANGES_REQUESTED** [review]

**Comment by Konicai** [comment]
Currently it looks like 
`abilityLayer.setLayerType(AbilityLayer.Type.SPECTATOR);` 
is simply overridden by the subsequent 
`abilityLayer.setLayerType(AbilityLayer.Type.BASE);`

And you're adding the exact same `abilityLayer` to the list in the packet twice.

This just looks like it reverts #3770 and honestly I'm skeptical if its an actual solution.

**Comment by patyhank** [comment]
It looks like #3770 can be revert, it doesn't seem to be a problem in the latest version.
This PR doesn't seem to be necessary anymore.

**Pull request closed without merging**